vyper/venom/basicblock.py

```diff
@@ -309,12 +341,108 @@ def is_pseudo(self) -> bool:
+        opcode = self.opcode
+        if opcode == "mstore":
+            if isinstance(self.operands[1], IRLiteral):
+                return MemoryLocation(offset=self.operands[1].value, size=32)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "mload":
+            return EMPTY_MEMORY_ACCESS
+        elif opcode == "mcopy":
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "calldatacopy":
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "dloadbytes":
+            return FULL_MEMORY_ACCESS
+        elif opcode == "dload":
+            return FULL_MEMORY_ACCESS
+        elif opcode == "invoke":
+            return FULL_MEMORY_ACCESS
+        elif opcode in ("call", "delegatecall", "staticcall"):
+            if isinstance(self.operands[1], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(
+                    offset=self.operands[1].value, size=self.operands[0].value, is_volatile=False
+                )
+            return FULL_MEMORY_ACCESS
+        elif opcode in ("codecopy", "extcodecopy"):
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "returndatacopy":
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        return EMPTY_MEMORY_ACCESS
+
+    def get_read_memory_location(self) -> MemoryLocation:
+        """Extract memory location info from an instruction"""
+        opcode = self.opcode
+        if opcode == "mstore":
+            return EMPTY_MEMORY_ACCESS
+        elif opcode == "mload":
+            if isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[0].value, size=32)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "mcopy":
+            if isinstance(self.operands[1], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[1].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "calldatacopy":
+            return EMPTY_MEMORY_ACCESS
+        elif opcode == "dloadbytes":
+            return EMPTY_MEMORY_ACCESS
+        elif opcode == "dload":
+            return EMPTY_MEMORY_ACCESS
```

Comment:
technically, dload writes to location 0

Reply:
Yes I know, just not taking the special memory location into account for this.